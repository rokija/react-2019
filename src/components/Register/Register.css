.Register {
    height: 100vh;
    background-color: rgb(243, 243, 243);

    display: flex;
    justify-content: center;
    align-items: center;
}

.Register__content {
    width: 30%;
    background-color: white;
    padding: 20px;
    border-radius: 10px;
}

.Register__content__link {
    margin-top: 10px;
}
