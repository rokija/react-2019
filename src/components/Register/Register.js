import React, { Component } from "react";
import { Button } from "reactstrap";
import { Link } from "react-router-dom";
import "./Register.css";

export const sum = (a, b) => a + b;

class Register extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      username: "",
      password: ""
    };
  }

  onInputChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    const { email, password, username } = this.state;

    return (
      <div className="Register">
        <div className="Register__content">
          <h2>Register</h2>
          {email}
          <form>
            <div className="form-group">
              <label>username</label>
              <input
                onChange={this.onInputChange}
                className="form-control"
                type="text"
                name="username"
                value={username}
              />
            </div>

            <div className="form-group">
              <label>email</label>
              <input
                onChange={this.onInputChange}
                className="form-control"
                type="email"
                name="email"
              />
            </div>

            <div className="form-group">
              <label>password</label>
              <input
                onChange={this.onInputChange}
                className="form-control"
                type="password"
                name="password"
                value={password}
              />
            </div>
            <Button
              onClick={() =>
                this.props.onRegister(
                  this.state.username,
                  this.state.email,
                  this.state.password
                )
              }
              color="dark"
            >
              Register
            </Button>
          </form>
          <div className="Register__content__link">
            <Link to="/login">already registered? login</Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Register;
