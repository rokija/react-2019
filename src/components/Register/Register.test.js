import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Register, { sum } from "./Register";

describe("Register component", () => {
  const component = shallow(<Register />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("changes state when onInputChange is called", () => {
    expect(component.state().email).toEqual("");

    component
      .instance()
      .onInputChange({ target: { name: "email", value: "mock value" } });

    expect(component.state().email).toEqual("mock value");
  });

  it("sum return 7 if given parameters are 3 and 4", () => {
    expect(sum(3, 4)).toEqual(7);
  });
});
