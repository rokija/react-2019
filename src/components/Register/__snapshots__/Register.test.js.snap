// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders properly 1`] = `
<div
  className="Register"
>
  <div
    className="Register__content"
  >
    <h2>
      Register
    </h2>
    <form>
      <div
        className="form-group"
      >
        <label>
          username
        </label>
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          email
        </label>
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          password
        </label>
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <Button
        color="dark"
        onClick={[Function]}
        tag="button"
      >
        Register
      </Button>
    </form>
    <div
      className="Register__content__link"
    >
      <Link
        replace={false}
        to="/login"
      >
        already registered? login
      </Link>
    </div>
  </div>
</div>
`;
